python/revng/cli/_commands/ptml.py

```diff
@@ -0,0 +1,273 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import argparse
+import re
+import sys
+from contextlib import suppress
+from typing import Any, Callable, Dict, List, Optional, TypeVar, Union
+from xml.dom import Node
+from xml.dom.minidom import Document, parseString
+
+import yaml
+
+from revng.cli.commands_registry import Command, Options, commands_registry
+
+try:
+    from rich.console import Console
+
+    color = True
+except ImportError:
+    color = False
+
+
+COLOR_CONVERSION = {
+    "asm.label": "bright_red",
+    "asm.label-indicator": "",
+    "asm.comment-indicator": "white",
+    "asm.mnemonic": "dodger_blue1",
+    "asm.mnemonic-prefix": "bright_green",
+    "asm.mnemonic-suffix": "bright_green",
+    "asm.immediate-value": "dark_slate_gray2",
+    "asm.memory-operand": "dodger_blue1",
+    "asm.register": "orange1",
+}
+
+
+def yaml_load(content: str) -> Dict[str, Any]:
+    loader = getattr(yaml, "CSafeLoader", yaml.SafeLoader)
+    return yaml.load(content, loader)
+
+
+def yaml_dump(obj: Any) -> str:
+    dumper = getattr(yaml, "CDumper", yaml.Dumper)
+    return yaml.dump(obj, Dumper=dumper)
+
+
+def normalize_filter_extract(filters: List[str], extract: Optional[str]) -> Union[str, List[str]]:
+    if extract is not None:
+        return extract
+    if len(filters) == 0:
+        return []
+    return ",".join(filters).split(",")
+
+
+def is_ptml(content: str) -> bool:
+    return bool(re.match(r"\s*<", content))
+
+
+T = TypeVar("T")
+
+
+def handle_file(
+    raw: str,
+    func_one: Callable[[str], T],
+    func_many: Callable[[Dict[str, str]], T],
+    filters: Union[str, List[str]],
+) -> T:
+    if is_ptml(raw):
+        if len(filters) > 0:
+            print("Cannot extract/filter a plain ptml file", file=sys.stderr)
```

Comment:
Never log to `stdout`

Reply:
Fixed, used `logging`

Reply:
Mmmh, `logging` sucks big time.
```
>>> import logging
>>> logging.warning("asd")
WARNING:root:asd
```
I see you introduced it in `revng-daemon`, but I'd prefer to stick with a simple `log` function as we're doing elsewhere.
If we decide to adopt something fancier, we need to adopt it *everywhere* and probably it won't be `logging`.

In order to avoid extending the scope of this PR, let's stick with a good old
```
def log(message):
    sys.stderr.write(message + "\n")
```
This will make refactoring way easier.

Reply:
```python
>>> import logging
>>> logging.basicConfig(format="%(message)s")
>>> logging.warning("asd")
asd
```
That said _sure_ the default isn't pretty, but besides that the `logging` library is pretty standard in terms of features (in fact it is used in `revng-daemon` since if we ever need to hook it to a LLVM logger it is _dead simple_)

I'll revert the commit and add `def log`, however I'd say to pick a logging library (I'd go with `logging` because it's in the standard library and pretty capable, but I'm open to using an external library) at some point, otherwise we'll end up with every module with their own `def log` that does things slightly differently and then it'll be a pain to switch.

Reply:
I'm in favor of switching, but it needs to be done globally. Opportunistic local improvements that deviate from the standard are a recipe for an incoherent code base that makes refactoring harder.
As a maintainer it's my duty to avoid this! `:)`.

I'd say spin up a new PR for this, but frankly we need to keep priorities straight and we already have a lot of infra projects in flight.